src/platform/plugins/shared/unified_doc_viewer/public/components/doc_viewer_table/table_cell_actions.tsx

```diff
@@ -66,6 +67,61 @@ const esqlMultivalueFilteringDisabled = i18n.translate(
   }
 );
 
+const Copy: React.FC<TableActionsProps & { toasts: IToasts }> = ({ Component, row, toasts }) => {
+  if (!row) {
+    return null;
+  }
+
+  const { formattedAsText, name } = row;
```

Comment:
Since `formattedAsText` is a custom getter with caching logic, it would be better to access it from `onClick` handler instead of when Copy component is rendered to prevent unnecessary computations.